bin/vim_erlang_tags.erl

```diff
@@ -120,77 +120,126 @@ get_command_type(C) when C=:=include; C=:=ignore; C=:=output -> stateful;
 get_command_type(B) when B=:=otp; B=:= verbose; B=:= help -> boolean.
 
 main(Args) ->
-    % Process arguments
-    put(files, []),
-    put(tagsfilename, "tags"),
-    put(ignored, []),
-    put(incl_otp, false),
-    parse_args(Args),
-    ArgFiles =
-        case get(files) of
-            [] ->
-                [?DEFAULT_PATH];
-            Other ->
-                Other
-        end,
-
-    Files =
-        case get(incl_otp) of
-            true -> [code:lib_dir()|ArgFiles];
-            false -> ArgFiles
-        end,
-
-    Tags = create_tags(Files),
-    ok = tags_to_file(Tags, get(tagsfilename)),
-    ets:delete(Tags).
-
-%% I know that using the process dictionary is not very nice...
-parse_args([]) ->
-    ok;
-parse_args(["-"|OtherArgs]) ->
-    put(files, [stdin|get(files)]),
-    parse_args(OtherArgs);
-parse_args([Help|_]) when Help == "-h";
-                          Help == "--help" ->
-    print_help(),
-    halt(0);
-parse_args([Verbose|OtherArgs]) when Verbose == "-v";
-                                     Verbose == "--verbose" ->
-    put(verbose, true),
-    log("Verbose mode on.~n"),
-    parse_args(OtherArgs);
-parse_args([InclOTP|OtherArgs]) when InclOTP == "-p";
-                                     InclOTP == "--otp" ->
-    put(incl_otp, true),
-    log("Including OTP in.~n"),
-    parse_args(OtherArgs);
-parse_args([Output, TagsFileName|OtherArgs]) when Output == "-o";
-                                                  Output == "--output" ->
-    put(tagsfilename, TagsFileName),
-    parse_args(OtherArgs);
-parse_args([Output]) when Output == "-o";
-                          Output == "--output";
-                          Output == "-i";
-                          Output == "--ignore" ->
-    log_error("More argument needed after ~s.~n", [Output]),
-    halt(1);
-parse_args([Ignored, Name|OtherArgs]) when Ignored == "-i";
-                                           Ignored == "--ignore" ->
-    Files = filelib:wildcard(Name),
-    AllIgnored = case get(ignored) of
-        undefined -> Files;
-        OldFiles -> OldFiles ++ [ filename:absname(N, ?DEFAULT_PATH) || N <- Files ]
+    log("Entering main. Args are ~p~n~n", [Args]),
+    ParsedArgs = reparse_args(?default_parsed_params, Args),
+    set_verbose_flag(ParsedArgs),
+    Opts = clean_opts(ParsedArgs),
+    run(Opts).
+
+run(#{help := true}) ->
+    print_help();
+run(#{explore := Explore, output := TagFile}) ->
+    EtsTags = create_tags(Explore),
+    ok = tags_to_file(EtsTags, TagFile),
+    ets:delete(EtsTags).
+
+set_verbose_flag(#{verbose := T}) ->
+    put(verbose, T),
+    log("Verbose mode on.~n").
+
+-spec reparse_args(parsed_params(), cmd_line_arguments()) -> parsed_params().
+reparse_args(Opts, []) ->
+    Opts;
+reparse_args(Opts, AllArgs) ->
+    {Param, ToContinueParsing} = parse_next_arg(AllArgs),
+    {ParamState, NextArgs} = case get_command_type(Param) of
```

Comment:
Following the de facto Erlang indentation (supported by Emacs and mostly supported by Vim and used in `vim_erlang_tags.erl`), this could be written either as:

    {ParamState, NextArgs} = case get_command_type(Param) of
                                boolean -> {true, ToContinueParsing};
                                stateful -> get_full_arg_state(
                                              Param,
                                              maps:get(Param, Opts, []),
                                              ToContinueParsing)
                             end,

Or (after adding a newline) as:

    {ParamState, NextArgs} =
        case get_command_type(Param) of
            boolean ->
                {true, ToContinueParsing};
            stateful ->
                get_full_arg_state(
                  Param, maps:get(Param, Opts, []), ToContinueParsing)
        end,

I prefer the second. (Unfortunately this is one of the few things that Vim gets wrong when indenting Erlang code. Also, I noticed that the old code didn't follow this convention.)